### ISSUE TITLE
Lighthouse returned error: DNS_FAILURE. DNS servers could not resolve the provided domain.

### ISSUE THREAD

**Issue by TarunGoel** [GoogleChrome/lighthouse#10459]
Did anyone experience this behavior and have any suggestions:
This is happening consistently with a prominent eCommerce site home page. We are trying to access the page using Google Page Speed Insights API and getting Lighthouse error.

500 Internal Server Error
{
  "code" : 500,
  "errors" : [ {
    "domain" : "lighthouse",
    "message" : "Lighthouse returned error: DNS_FAILURE. DNS servers could not resolve the provided domain.",
    "reason" : "lighthouseError"
  } ],
  "message" : "Lighthouse returned error: DNS_FAILURE. DNS servers could not resolve the provided domain."
}
com.google.api.client.googleapis.json.GoogleJsonResponseException: 500 Internal Server Error
{
  "code" : 500,
  "errors" : [ {
    "domain" : "lighthouse",
    "message" : "Lighthouse returned error: DNS_FAILURE. DNS servers could not resolve the provided domain.",
    "reason" : "lighthouseError"
  } ],
  "message" : "Lighthouse returned error: DNS_FAILURE. DNS servers could not resolve the provided domain."
}

**Comment by connorjclark** [comment]
URL?

**Comment by TarunGoel** [comment]
@connorjclark - https://www.ulta.com/

**Comment by TarunGoel** [comment]
Please help in here and suggest what could be a issue here.

**Comment by patrickhulce** [comment]
I'm not able to reproduce this error on either web.dev/measure or PSI frontend. Perhaps it was just an intermittent failure that's resolved itself?

![image](https://user-images.githubusercontent.com/2301202/77660043-d3cc5900-6f46-11ea-8796-d11a6752b9d3.png)

![image](https://user-images.githubusercontent.com/2301202/77660054-d7f87680-6f46-11ea-9fe3-e18634c14b72.png)

**Comment by rviscomi** [comment]
I'm running into this issue using the PSI API as well. It's specifically when the PWA category is selected and only for certain URLs. For example:

`https://www.googleapis.com/pagespeedonline/v5/runPagespeed?url=https%3A%2F%2Fwww.cbs.com%2F&key=API_KEY&category=PWA`

```
{
  "error": {
    "code": 500,
    "message": "Lighthouse returned error: DNS_FAILURE. DNS servers could not resolve the provided domain.",
    "errors": [
      {
        "message": "Lighthouse returned error: DNS_FAILURE. DNS servers could not resolve the provided domain.",
        "domain": "lighthouse",
        "reason": "lighthouseError"
      }
    ]
  }
}
```

However the API returns expected results when changing the URL to `https://www.example.com/`.

cc @exterkamp

**Comment by csvan** [comment]
Having the same issue, but only when running headless Chrome. Standard settings work fine for the same URL. I am almost certain it has something to do with the company proxy, but not sure what.

**Comment by zdelagrange** [comment]
I am having the same issue and believe it to be due to networking. Accessing www sites works fine, but internal network sites result in the DNS error. Were you able to find anything @csvan ?

**Comment by csvan** [comment]
@zdelagrange unfortunately no, but I suspect that Chrome headless treats system proxies differently than the headed one (I am behind a corporate proxy). I don't know exactly where the rub lies unfortunately.

**Comment by zdelagrange** [comment]
The underlying issue for me ended up being running inside a container that
didn’t have a large enough /dev/shm. By mounting the local file systems
/dev/shm to the containers /dev/shm we were able to get stable results

**Comment by mslowiak** [comment]
I have the same problem.
Using internal openshift cluster dns and headless option for lighthouse

**Comment by diego-betto** [comment]
same here. Javascript version not working with or without API key. Curl is ok.

Try this version 

```html
<script>
        const apiKey = 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'; // added
        const url = 'https://yourwebsite.tld' // added

        function getDisplayValue(obj, key) { // added
            return obj && obj[key] && obj[key].displayValue || null; // added
        } // added

        function run() {
            const url = setUpQuery();
            fetch(url)
                .then(response => response.json())
                .then(json => {
                    // See https://developers.google.com/speed/docs/insights/v5/reference/pagespeedapi/runpagespeed#response
                    // to learn more about each of the properties in the response object.
                    showInitialContent(json.id);
                    const cruxMetrics = {
                        "First Contentful Paint": json.loadingExperience?.metrics?.FIRST_CONTENTFUL_PAINT_MS?.category || '', // changed
                        "First Input Delay": json.loadingExperience?.metrics?.FIRST_INPUT_DELAY_MS?.category || '' // changed
                    };
                    showCruxContent(cruxMetrics);
                    const lighthouse = json.lighthouseResult;
                    const lighthouseAudits = lighthouse?.audits; // added

                    const lighthouseMetrics = {
                        'First Contentful Paint': getDisplayValue(lighthouseAudits, 'first-contentful-paint'), // changed
                        'Speed Index': getDisplayValue(lighthouseAudits, 'speed-index'), // changed
                        'Time To Interactive': getDisplayValue(lighthouseAudits, 'interactive'), // changed
                        'First Meaningful Paint': getDisplayValue(lighthouseAudits, 'first-meaningful-paint'), // changed
                        'First CPU Idle': getDisplayValue(lighthouseAudits, 'first-cpu-idle'), // changed
                        'Estimated Input Latency': getDisplayValue(lighthouseAudits, 'estimated-input-latency'), // changed
                    };
                    showLighthouseContent(lighthouseMetrics);
                });
        }

        function setUpQuery() {
            const api = 'https://www.googleapis.com/pagespeedonline/v5/runPagespeed';
            const parameters = {
                url: encodeURIComponent(url),
                key: apiKey // added
            };
            let query = [`${api}?`]; // changed
            for (key in parameters) {
                query.push(`${key}=${parameters[key]}`); // changed
            }
            return query.join('&'); // changed
        }

        function showInitialContent(id) {
            document.body.innerHTML = '';
            const title = document.createElement('h1');
            title.textContent = 'PageSpeed Insights API Demo';
            document.body.appendChild(title);
            const page = document.createElement('p');
            page.textContent = `Page tested: ${id}`;
            document.body.appendChild(page);
        }

        function showCruxContent(cruxMetrics) {
            const cruxHeader = document.createElement('h2');
            cruxHeader.textContent = "Chrome User Experience Report Results";
            document.body.appendChild(cruxHeader);
            for (key in cruxMetrics) {
                const p = document.createElement('p');
                p.textContent = `${key}: ${cruxMetrics[key]}`;
                document.body.appendChild(p);
            }
        }

        function showLighthouseContent(lighthouseMetrics) {
            const lighthouseHeader = document.createElement('h2');
            lighthouseHeader.textContent = "Lighthouse Results";
            document.body.appendChild(lighthouseHeader);
            for (key in lighthouseMetrics) {
                const p = document.createElement('p');
                p.textContent = `${key}: ${lighthouseMetrics[key]}`;
                document.body.appendChild(p);
            }
        }

        run();
    </script>
```

**Comment by WywySenarios** [comment]
> ```
> let query = [`${api}?`]; // changed
>             for (key in parameters) {
>                 query.push(`${key}=${parameters[key]}`); // changed
>             }
>             return query.join('&'); // changed
> ```

I got the same DNS failure error too, and I think the reason why is because the querystring is invalid.

Your code's "setUpQuery" function will return something like: "https://www.googleapis.com/pagespeedonline/v5/runPagespeed?&url=[insert url name]&key[insert api key]"

The query string is invalid because there's an extra "&" right after the "?"

When I modified my own code to ask for the url with a valid querystring, there seemed to be no error. Could it be that this error involves the querystring and the server should catch this error and return a meaningful result?
